code/object/objectdock.cpp

```diff
@@ -708,6 +708,27 @@ void dock_find_max_speed_helper(object *objp, dock_function_info *infop)
 		infop->maintained_variables.objp_value = objp;
 	}
 }
+
+void object_set_arriving_stage1_ndl_flag_helper(object *objp, dock_function_info * /*infop*/ )
+{
+	Ships[objp->instance].flags.set(Ship::Ship_Flags::Arriving_stage_1_ndl);
```

Comment:
This will set the flag on all ships in the docked group.  Do you want to assign it to all ships *except* the leader?  If so, add a check for the ship with the leader flag.

Reply:
I considered excluding the leader and thought it doesn't actually matter. However as you've raised above it's probably _clearer_ to the coder if it's excluded; and that's worthwhile so I'll change it.